### PR TITLE
[v0.5] s4: Fixes 227

### PR DESCRIPTION
**Pull request by pmatseykanets**
## Issue: 

Port of #390

* Validate user retention userattribute fields and settings
* [v0.4.s4] Don't allow empty lastLogin user attribute

## CheckList
  
- [ ] Test
  
- [ ] Docs

### CODE REVIEW

**Review by ericpromislow — CHANGES_REQUESTED** [review]
The PR fails because the drone config uses `rancher/rancher:v2.9-head` and that image quietly changed its base image from `registry.suse.com/bci-base` to `bci-micro`, and the latter doesn't include `zypper` .

I fixed this on other PRs that still use drone by moving it back to `v2.9.0-alpha1`

**Comment by pmatseykanets** [comment]
@ericpromislow Thanks for the insights! It seems like the required change is not related to this port in any way and should be applied is a separate PR. Then this one can be rebased.

**Comment by pmatseykanets** [comment]
Depends on and needs to be rebased off of #406

**Review by JonCrowther — APPROVED** [review]

**Review by raulcabello — APPROVED** [review]

File: pkg/resources/management.cattle.io/v3/setting/Setting.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+## Validation Checks
+
+### Invalid Fields - Create
+
+When a Setting is created, the following checks take place:
+
+- If set, `disable-inactive-user-after` must be zero or a positive duration (e.g. `240h`).
+- If set, `delete-inactive-user-after` must be zero or a positive duration (e.g. `240h`).
```
Comment (andreas-kupries) [inline pkg/resources/management.cattle.io/v3/setting/Setting.md:8]: The `MinDeleteInactiveUserAfter` check does not seem to be documented. Is that intentional ?

**Pull request merged**